src/claims/claims.controller.ts

```diff
@@ -105,18 +121,58 @@ export class ClaimsController {
   }
 
   @ApiOperation({
-    summary: 'Get CustomClaimType',
-    description:
-      'This endpoint will retrieve the custom claim type by id or name. <br /> ID or name needs to be provided. <br/> If both are provided will fetch using ID',
+    summary: 'Get CustomClaimTypes',
+    description: 'This endpoint fetches CustomClaimTypes that have been registered."',
+  })
+  @Get('custom-claim-types')
+  async getCustomClaimTypes(
+    @Query() { size, start, dids }: GetCustomClaimTypesDto
+  ): Promise<PaginatedResultsModel<CustomClaimTypeWithDid>> {
+    const { data, count, next } = await this.claimsService.getRegisteredCustomClaimTypes(
+      size,
+      new BigNumber(start || 0),
+      dids
+    );
+
+    return new PaginatedResultsModel({
+      results: data.map(claimType => new CustomClaimTypeWithDid(claimType)),
+      total: count,
+      next,
+    });
+  }
+
+  @ApiOperation({
+    summary: 'Get CustomClaimType by ID',
+    description: 'This endpoint fetches the CustomClaimType, identified by its ID',
   })
-  @ApiQuery({
+  @ApiParam({
     name: 'id',
     description: 'The ID of the CustomClaimType',
     type: 'string',
     required: false,
     example: '1',
   })
-  @ApiQuery({
+  @ApiNotFoundResponse({
+    description: 'The CustomClaimType does not exist',
+  })
+  @Get('custom-claim-types/id/:id')
```

Comment:
```suggestion
  @Get('custom-claim-types/:id')
```

Seems weird to have `/id` in the route to me. I see its to distringuish between name though.

If name is always unique I think we could have a single endpoint that would return details for either.

```
  @Get('custom-claim-types/:id')
  
  if(id instanceof BigNumber) { 
    this.claimsService.getCustomClaimTypeById(id);
   else {
     this.claimsService.getCustomClaimTypeByName(name);
   }
  ```
  
  This might make the validators a bit tricky though.